spec/unit/pops/types/error_spec.rb

```diff
@@ -0,0 +1,191 @@
+          notice(Error[default,'a'])
+          notice(Error[default,/a/])
+          notice(Error[default,Enum['a', 'b']])
+          notice(Error[default,Pattern[/a/, /b/]])
+          notice(Error['a','a'])
+          notice(Error[/a/,/a/])
+          notice(Error[Enum['a', 'b'],Enum['a', 'b']])
+          notice(Error[Pattern[/a/, /b/],Pattern[/a/, /b/]])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq([
+              "Error['a']",
+              'Error[/a/]',
+              "Error[Enum['a', 'b']]",
+              "Error[Pattern[/a/, /b/]]",
+              "Error['a']",
+              'Error[/a/]',
+              "Error[Enum['a', 'b']]",
+              "Error[Pattern[/a/, /b/]]",
+              "Error[default, 'a']",
+              'Error[default, /a/]',
+              "Error[default, Enum['a', 'b']]",
+              "Error[default, Pattern[/a/, /b/]]",
+              "Error['a', 'a']",
+              'Error[/a/, /a/]',
+              "Error[Enum['a', 'b'], Enum['a', 'b']]",
+              "Error[Pattern[/a/, /b/], Pattern[/a/, /b/]]",
+            ])
+          end
+        end
+
+        context 'a Error instance' do
+          it 'can be created from a string' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error')
+            notice($o)
+            notice(type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq([
+              "Error({'message' => 'bad tings happened', 'kind' => 'puppet/error'})",
+              "Error['puppet/error', 'ERROR']"
+            ])
+          end
+
+          it 'can be created from a hash' do
+            code = <<-CODE
+            $o = Error(message => 'Sorry, not implemented', kind => 'puppet/error', issue_code => 'NOT_IMPLEMENTED')
+            notice($o)
+            notice(type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq([
+              "Error({'message' => 'Sorry, not implemented', 'kind' => 'puppet/error', 'issue_code' => 'NOT_IMPLEMENTED'})",
+              "Error['puppet/error', 'NOT_IMPLEMENTED']"
+            ])
+          end
+
+          it 'is an instance of its type' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error')
+            notice($o =~ type($o))
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching kind' do
+            code = <<-CODE
+            $o = Error('bad tings happened', 'puppet/error')
+            notice($o =~ Error[/puppet\\/error/])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
+          end
+
+          it 'is an instance of matching default issue' do
+            code = <<-CODE
+            $o = Error('bad tings happened')
+            notice($o =~ Error[default, 'ERROR'])
+            CODE
+            expect(eval_and_collect_notices(code)).to eq(['true'])
```

Comment:
I don't think 'ERROR' should be default issue_code, and this test therefore needs to change. (They are all errors so it is tautological).